flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/State.java

```diff
@@ -43,7 +43,14 @@ public State(final String name, final StateType stateType) {
 		this.name = name;
 		this.stateType = stateType;
 
-		stateTransitions = new ArrayList<StateTransition<T>>();
+		stateTransitions = new ArrayList<>();
+	}
+
+	public State(String name, StateType stateType, Collection<StateTransition<T>> stateTransitions) {
```

Comment:
I agree with @chermenin and this class can be reverted to its previous state. In general, PRs should have the smallest diff possible in order to be easier to review.